### DISCUSSION TITLE
cookie not deleting in load function?

### DISCUSSION THREAD

**Discussion by manstie** [sveltejs/kit#6872] [General]
I have recently migrated from `1.0.0-next.449` to `1.0.0-next.484`.

I have a `/logout` endpoint with a `+page.server.js` and a `+page.svelte`. In the `server` I changed it from:
```
/** @type {import('./$types').PageServerLoad} */
export function load({ locals, setHeaders }) {
	...
	setHeaders({
		'set-cookie': 'jwt=deleted; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT',
	});
	return {};
}
```
to
```
/** @type {import('./$types').PageServerLoad} */
export function load({ locals, cookies }) {
	...
	cookies.delete('jwt');
	return {};
}
```
Much nicer, but now it doesn't actually set / delete the cookie when the svelte navigation goes to that page. I had to add `rel="external"` to my logout navigation button to force sveltekit to hard refresh and run the load function or something? What's up with that.

**Comment by teemingc** [comment]
You'll have to add the `path` option in `cookies.delete`, so:

```ts
cookies.delete('jwt', { path: '/' });
```

See [this issue](https://github.com/sveltejs/kit/issues/6609#issuecomment-1247425220) and [this PR](https://github.com/sveltejs/kit/pull/6811#discussion_r971199741) for some insight into why `cookies.delete` does not use `/` as a default, and why it's needed to delete certain cookies.

**Reply by manstie** [reply]
That works, thanks.

**Reply by DoisKoh** [reply]
> You'll have to add the `path` option in `cookies.delete`, so:
> 
> ```ts
> cookies.delete('jwt', { path: '/' });
> ```
> 
> See [this issue](https://github.com/sveltejs/kit/issues/6609#issuecomment-1247425220) and [this PR](https://github.com/sveltejs/kit/pull/6811#discussion_r971199741) for some insight into why `cookies.delete` does not use `/` as a default, and why it's needed to delete certain cookies.

Man... I wasted so much time on this. Thanks so much.

**Reply by yousufiqbal** [reply]
It's working for me with plain `cookies.delete('cookie-name')` without setting `path` option.

**Reply by DoisKoh** [reply]
For me, adding the path made it work on my local server, but not on my development servers. I'm not sure what added security caused this to be an issue, but after specifying most of the options (sameSite, httpOnly, and secure), it worked.
